packages/page-claims/src/Attest.tsx

```diff
@@ -0,0 +1,109 @@
+// Copyright 2017-2020 @polkadot/app-claims authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import React, { useEffect, useCallback, useState } from 'react';
+import styled from 'styled-components';
+import { Button, Card, TxButton } from '@polkadot/react-components';
+import { useApi } from '@polkadot/react-hooks';
+import { FormatBalance } from '@polkadot/react-query';
+import { Option } from '@polkadot/types';
+import { BalanceOf, EthereumAddress, StatementKind } from '@polkadot/types/interfaces';
+
+import Statement from './Statement';
+import { useTranslation } from './translate';
+import { getStatementSentence } from './util';
+
+interface Props {
+  accountId: string;
+  className?: string;
+  ethereumAddress: EthereumAddress | null;
+  statementKind?: StatementKind;
+}
+
+function Attest ({ accountId, className, ethereumAddress, statementKind }: Props): React.ReactElement<Props> | null {
+  const { t } = useTranslation();
+  const { api } = useApi();
+  const [claimValue, setClaimValue] = useState<BalanceOf | null>(null);
+  const [claimAddress, setClaimAddress] = useState<EthereumAddress | null>(null);
+  const [isBusy, setIsBusy] = useState(false);
+
+  const _fetchClaim = useCallback(
```

Comment:
Not sure why this is a callback. Extract the function, pass the api, address, flag setter in from where it is called. (Or keep inside effect)

Reply:
Actually copied from https://github.com/polkadot-js/apps/blob/master/packages/page-claims/src/Claim.tsx#L30, thought that was the way you did it in apps.